cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -155,25 +180,30 @@ private AccessToken acquireAccessToken(ClientConfig clientConfig, ConnectionConf
     }
 
     try {
-      AccessToken savedAccessToken = getSavedAccessToken(connectionInfo.getHostname());
-      checkConnection(clientConfig, connectionInfo, savedAccessToken);
-      return savedAccessToken;
+      UserAccessToken savedToken = getSavedAccessToken(connectionInfo.getHostname());
+      connectionInfo = new AuthenticatedConnectionConfig(connectionInfo, savedToken.getUsername());
+      clientConfig.setConnectionConfig(connectionInfo);
+      checkConnection(clientConfig, savedToken.getAccessToken());
+      return savedToken.getAccessToken();
     } catch (UnauthorizedException ignored) {
       // access token invalid - fall through to try acquiring token manually
     }
 
-    return getNewAccessToken(connectionInfo, output, debug);
+    return getNewAccessToken(clientConfig, output, debug);
```

Comment:
Instead of setting `clientConfig`'s connection, you should simply return `UserAccessToken` instead of returning `AccessToken`. Then, in `tryConnect`, you'll call `checkConnection` which will create your `AuthenticatedConnectionConfig` and returns it.